lib/fluent/parser.rb

```diff
@@ -381,6 +383,12 @@ def parse(text)
         text.split(delimiter).each do |pair|
           key, value = pair.split(label_delimiter, 2)
           @keys.push(key)
+          if null_empty_string
+            value = (value == '') ? nil : value
+          end
+          if null_value_pattern
+            value = (value =~ /#{null_value_pattern}/) ? nil : value
```

Comment:
Creating regexp object at each record is high cost.
It should be done in configure method.